contracts/zero-ex/contracts/src/features/UniswapV2ForkFeature.sol

```diff
@@ -0,0 +1,205 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/contracts/src/v06/IERC20TokenV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/IEtherTokenV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/LibSafeMathV06.sol";
+import "../migrations/LibMigrate.sol";
+import "../fixins/FixinCommon.sol";
+import "../fixins/FixinTokenSpender.sol";
+import "./interfaces/IFeature.sol";
+import "./interfaces/IUniswapV2ForkFeature.sol";
+import "./interfaces/IUniswapV2ForkPair.sol";
+
+/// @dev VIP UniswapV2Fork  fill functions.
+contract UniswapV2ForkFeature is
+    IFeature,
+    IUniswapV2ForkFeature,
+    FixinCommon,
+    FixinTokenSpender
+{
+    using LibSafeMathV06 for uint256;
+
+    /// @dev Name of this feature.
+    string public constant override FEATURE_NAME = "UniswapV2ForkFeature";
+    /// @dev Version of this feature.
+    uint256 public immutable override FEATURE_VERSION = _encodeVersion(1, 0, 0);
+
+    address private constant ETH_ADDRESS = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    address private constant WETH_ADDRESS = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+
+    // TODO: complete the list
+    // address of the SushiSwapV2 factory contract.
+    address constant private SUSHISWAPV2_FACTORY = address(0xc35DADB65012eC5796536bD9864eD8773aBc74C4);
+
+    function sellToUniswapV2Fork(
+        ProtocolFork fork,
+        IERC20TokenV06[] calldata tokens,
+        uint256 sellAmount,
+        uint256 minBuyAmount
+    )
+        external
+        payable
+        override
+        returns (uint256 buyAmount)
+    {
+        require(tokens.length > 1, 'UniswapV2ForkFeature/INVALID_TOKENS_LENGTH');
+
+        IERC20TokenV06[] memory path = tokens;
+        if (address(tokens[0]) == ETH_ADDRESS) {
+            IEtherTokenV06(WETH_ADDRESS).deposit{value:msg.value}();
+            path[0] = IERC20TokenV06(WETH_ADDRESS);
+        }
+
+        bool unwrapWeth = address(tokens[tokens.length - 1]) == ETH_ADDRESS;
+        if (unwrapWeth) {
+            path[path.length - 1] = IERC20TokenV06(WETH_ADDRESS);
+        }
+
+        address factory = _getFactory(fork);
+        uint[] memory amounts = _getAmountsOut(factory, sellAmount, path);
+        buyAmount = amounts[amounts.length - 1];
+        require(buyAmount >= minBuyAmount, 'UniswapV2ForkFeature/INSUFFICIENT_OUTPUT_AMOUNT');
+        _transferERC20TokensFrom(
+            path[0],
+            msg.sender,
+            _pairFor(factory, path[0], path[1]),
+            amounts[0]
+        );
+        _swap(amounts, path, path[path.length - 1], factory);
```

Comment:
What if `_swap()` computed `amounts` on the fly? iteratively? I think we can avoid having to do a `_pairFor()` 2+ times per pool if we do it that way.

Reply:
Also, why is `_to` being set to a token?

Remeber we want the first recipient to be:
1. The next pool if `path.length > 2`
2. The taker if `!unwrapWeth`
3. Otherwise `this`

Reply:
getting I stack too deep when I put all together, I have to split the swap into two, lmk if that make sense.

Reply:
addressed the recipient issue. thx